File: 005-Largest-palindrome-product/main.py
# Finding the largest palindrome product.
def is_palindrome(n):
    return str(n) == str(n)[::-1]

def largest_palindrome_product():
    largest_palindrome = 0
    for x in range(100, 1000):
        for y in range(100, 1000):
            z = x * y
            if is_palindrome(z) and z > largest_palindrome:
                largest_palindrome = z
    return largest_palindrome

result = largest_palindrome_product()
print("The largest palindrome: ", result)

File: 005-Largest-palindrome-product/main.cpp
#include <iostream>
#include <algorithm>

using namespace std;

bool is_palindrome(int n) {
    string num_str = to_string(n);

    string reversed_str = num_str;
    reverse(reversed_str.begin(), reversed_str.end());

    return num_str == reversed_str;
}

int largest_palindrome_product() {

    int largest_palindrome = 0;

    for(int i = 100; i < 1000; i++) {
        for(int y = 100; y < 1000; y++) {
            int z = i * y;
            if(is_palindrome(z) && z > largest_palindrome) {
                largest_palindrome = z;
            }
        }
    }
    return largest_palindrome;
}

int main() {

    int result = largest_palindrome_product();

    cout << "The largest palindrome is: " << result << endl;

    return 0;
}